tests/components/kodi/__init__.py

```diff
@@ -11,13 +11,24 @@
     CONF_USERNAME,
 )
 
-from .util import MockConnection
+from .util import MockConnection, MockWSConnection
 
 from tests.common import MockConfigEntry
 
+PATCH_KODI = "homeassistant.components.kodi"
+PATCH_KODI_BINARY = f"{PATCH_KODI}.binary_sensor.KodiBinaryEntity"
+PATCH_KODI_CONNMAN = f"{PATCH_KODI}.kodi_connman.KodiConnectionManager"
+PATCH_KODI_MEDIA = f"{PATCH_KODI}.media_player"
 
-async def init_integration(hass) -> MockConfigEntry:
-    """Set up the Kodi integration in Home Assistant."""
+
+async def init_integration(
```

Comment:
This should probably be a fixture. Also, why the underscores before the variable names?

Reply:
> why the underscores before the variable names

This is a leftover. I will remove this with the next commit.

> This should probably be a fixture.

First, I just used `init_integration` because it was already there and also used in the test_init.py file. Considering what this function now does, it seams more practical to stick with this function. What I looked at:

* Parameter to choose `use_ws` and `connected`
  Could be achieved with [marker](https://docs.pytest.org/en/6.2.x/fixture.html#using-markers-to-pass-data-to-fixtures) but is not as straight forward imho.
* Pass the `hass` object from the test function to it
  That I couldn't figure out. Maybe somehow like [this](https://stackoverflow.com/a/28570677).
  
```Python
@pytest.fixture()
def init_integration():
    def _init_integration(hass):
        # doing init stuff
        return entry

    return _init_integration

def test_something(hass,init_integration):
    entry = init_integration(hass)
    # doing tests
```
As I said, I don't see advantages in this case.

Reply:
I think you should have two separate fixtures - one for setting up with ws, and one without. As for `connected`, I don't think it should be part of this fixture - it can be a separate one.
You can see for example what I did in the Risco integration: https://github.com/home-assistant/core/blob/dev/tests/components/risco/conftest.py (if you create such a file with the same name for this integration the fixtures will automatically be included - no need to import them)

Reply:
I also learned all of this in a previous review when I was on your side :)

Reply:
Ok, I'll try to get rid of the `init_integration` function and turn it into fixtures.

But, why the hack can every function (also fixtures) just have the `hass` argument, with the correct instance?

Reply:
Ah, a fixture itself 🤦🏼‍♂️, of course. Enough for today. I still need to befriend the idea of having 6 fixtures (entry ws, entry http, connection ws online/offline, connection http online/offline) instead of one function. Maybe I find some kind of compromise.

Reply:
I created a fixture as you said, but I'm using marker for the connection state. Otherwise there would be too much duplicated code. I hope you can agree in this case?